recipes/zbar/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+               "with_video": [True, False],
+               "with_imagemagick": [True, False],
+               "with_gtk": [True, False],
+               "with_qt": [True, False],
+               "with_python_bindings": [True, False],
+               "with_x": [True, False],
+               "with_xshm": [True, False],
+               "with_xv": [True, False],
+               "with_jpeg": [True, False],
+               "enable_pthread": [True, False],}
+    default_options = {'shared': False,
+                       'fPIC': True,
+                       'with_video': False,
+                       'with_imagemagick': False,
+                       'with_gtk': False,
+                       'with_qt': False,
+                       'with_python_bindings': False,
+                       'with_x': False,
+                       'with_xshm': False,
+                       'with_xv': False,
+                       'with_jpeg': False,
+                       'enable_pthread': True,
+                       }
+
+    _env_build = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_autotools(self):
+        if not self._env_build:
+            self._env_build = AutoToolsBuildEnvironment(self)
+            env_args = []
+            env_args.extend([
+                "--enable-video" if self.options.with_video else "--disable-video",
+                "--with-imagemagick" if self.options.with_imagemagick else "--without-imagemagick",
+                "--with-gtk" if self.options.with_gtk else "--without-gtk",
+                "--with-qt" if self.options.with_qt else "--without-qt",
+                "--with-python" if self.options.with_python_bindings else "--without-python",
+                "--with-x" if self.options.with_x else "--without-x",
+                "--with-xshm" if self.options.with_xshm else "--without-xshm",
+                "--with-xv" if self.options.with_xv else "--without-xv",
+                "--with-jpeg" if self.options.with_jpeg else "--without-jpeg",
+                "--enable-pthread" if self.options.enable_pthread else "--disable-pthread",
+                "--enable-shared" if self.options.shared else "--disable-shared",
+                "--enable-static" if not self.options.shared else "--disable-static",
+            ])
+            self._env_build.configure(args=env_args, configure_dir=self._source_subfolder)
+        return self._env_build
+
+    def requirements(self):
+        if self.options.with_jpeg:
+            self.requires.add("libjpeg/9d")
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Zbar can't be built on Windows")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        env_build = self._configure_autotools()
```

Comment:
@ericriff Can you try this? Maybe we are seeing undefined behavior because of the initial `_`.
```suggestion
    def _patch_souces(self):
        tools.replace_in_file(os.path.join(self._source_subfolder, "zbar", "refcnt.c"), "_zbar_reflock", "zbar_reflock")
        tools.replace_in_file(os.path.join(self._source_subfolder, "zbar", "refcnt.h"), "_zbar_reflock", "zbar_reflock")

    def build(self):
        self._patch_souces()
        env_build = self._configure_autotools()
```

Reply:
(Untested)

Reply:
Sure I'll test it

Reply:
Maybe rename the function to _patch_sou**r**ces :smile:

Reply:
I'll fix it it the build succeeds :)

Reply:
Should I try to remove the underscore from the other objects? I don't really know whats going on haha

Reply:
It was just a guess. A bad guess, so it looks...

Reply:
I've been reading and that underscore at the beginning of the object name is needed on mac for extern symbols

Reply:
Could you please file an issue to the author? It seems be an upstream's bug

Reply:
I can and I will, but it seems like the project has been abandoned. 
I think the best approach in this case would be to raise an invalid config exception for macOS+static.